packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,151 @@
+import { Construct, Duration } from '@aws-cdk/core';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from '@aws-cdk/custom-resources';
+import { arnForDeploymentConfig } from '../utils';
+import { ILambdaDeploymentConfig } from './deployment-config';
+
+/**
+ * Lambda Deployment config type
+ */
+export enum CustomLambdaDeploymentConfigType {
+  /**
+   * Canary deployment type
+   */
+  CANARY = 'Canary',
+
+  /**
+   * Linear deployment type
+   */
+  LINEAR = 'Linear'
+}
+
+/**
+ * Properties of a reference to a CodeDeploy Lambda Deployment Configuration.
+ */
+export interface CustomLambdaDeploymentConfigProps {
+
+  /**
+   * The type of deployment config, either CANARY or LINEAR
+   */
+  readonly type: CustomLambdaDeploymentConfigType;
+
+  /**
+   * The integer percentage of traffic to shift:
+   * - For LINEAR, the percentage to shift every interval
+   * - For CANARY, the percentage to shift until the interval passes, before the full deployment
+   */
+  readonly percentage: number;
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: Duration;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ *
+ */
+export class CustomLambdaDeploymentConfig extends Construct implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    this.validateParameters(props);
+
+    const deploymentType = 'TimeBased' + props.type.toString();
+    const intervalMinutes = props.interval.toMinutes().toString();
+    const percentage = props.percentage.toString();
+
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + percentage+ 'Percent' + intervalMinutes + 'Minutes';
```

Comment:
Interpolation I think is better here:

```suggestion
      customDeploymentConfigName = `LambdaCanary${percentage}Percent${intervalMinutes}Minutes`;
```

Reply:
Done